sentry-jmx/src/main/java/io/sentry/jmx/GlobalSentryConfiguration.java

```diff
@@ -0,0 +1,30 @@
+package io.sentry.jmx;
+
+import io.sentry.Sentry;
+import io.sentry.SentryClient;
+import javax.management.InstanceAlreadyExistsException;
+import javax.management.MBeanRegistrationException;
+import javax.management.MBeanServer;
+import javax.management.MalformedObjectNameException;
+import javax.management.NotCompliantMBeanException;
+import javax.management.ObjectName;
+
+public class GlobalSentryConfiguration extends AbstractSentryConfiguration {
+  @Override
+  protected SentryClient getClient() {
+    return Sentry.getStoredClient();
+  }
+
+  @Override
+  public void reinitialize(String dsn) {
+    Sentry.init(dsn);
```

Comment:
This is where it gets a little weird. We've never made the assumption that people would be replacing the globally stored client:

https://github.com/getsentry/sentry-java/blob/ffd9fb6fc1f56815bdb11418d522b2599de34c27/sentry/src/main/java/io/sentry/Sentry.java#L132-L133

If people actually want/need to do this we probably have to consider what it it implies in regards to cleanup. For example calling `closeConnection` would at least shutdown the asyn connection's threadpool:

https://github.com/getsentry/sentry-java/blob/ffd9fb6fc1f56815bdb11418d522b2599de34c27/sentry/src/main/java/io/sentry/SentryClient.java#L233

Reply:
Yeah, this will probably require a policy decision if it fits into the supported usecases.